test/conformance/gateway_conformance_test.go

```diff
@@ -24,12 +24,23 @@ import (
 var skippedTestsForTraditionalRoutes = []string{
 	// core conformance
 	tests.HTTPRouteHeaderMatching.ShortName,
+	// Kong gateway does not support this feature
```

Comment:
This is quite sad, as it means we cannot claim GRPC conformance. Can you provide some details on why this fails?

Reply:
I will provide a reproduction of it and offer more information.

Reply:
An issue has already been created to fix the skipped case here.  I would like to merge this PR first.
https://github.com/Kong/kubernetes-ingress-controller/issues/6136


For the expression route, it fails to match due to priority issues.
https://kongstrong.slack.com/archives/C03CTMSHP6C/p1717569907597749